src/Extractor/ResponseRpcExtractor.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\JsonRpc\Extractor;
+
+use Psr\Http\Message\ResponseInterface;
+
+final class ResponseRpcExtractor extends RpcExtractor
+{
+    public static function getRpcErrorCode(ResponseInterface $response) : ?int
+    {
+        $error = self::getFieldFromBody($response, 'error');
+
+        return $error['code'] ?? null;
+    }
+
+    public static function getRpcErrorMessage(ResponseInterface $response) : ?string
+    {
+        $error = self::getFieldFromBody($response, 'error');
+
+        return $error['message'] ?? null;
+    }
+
+    /**
+     * @return array<array<array<string>>>|null
```

Comment:
mixed|null

Reply:
nah. proc

btw mixed|null = mixed

Reply:
vím že technicky `mixed|null = mixed`, ale často mixed znamená prostě int, string, array, ... null je tak trochu special case a imo si to zaslouží explicitní vyjmenování :)

no a mixed proto, protože to prostě je mixed :D v data může být cokoliv, ne jen tonhle konkrétní pole co jsi poslal

Reply:
ee to je confusing, kdyz pak delas treba s phpstanem. se to nauci kazdej pak blbe

koukam

Reply:
confusing mi to nepřijde ... ale jak chceš, opět na to netrvám :)